File: app/src/main/java/ir/arapp/arappmain/View/Activities/SplashScreenActivity.java
package ir.arapp.arappmain.View.Activities;

import androidx.annotation.RequiresApi;
import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;
import android.content.Intent;
import android.os.Build;
import android.os.Bundle;
import android.os.Handler;
import android.view.View;
import ir.arapp.arappmain.R;
import ir.arapp.arappmain.Util.Services.SessionManager;
import ir.arapp.arappmain.databinding.ActivitySplashScreenBinding;

public class SplashScreenActivity extends AppCompatActivity
{

//    region Variables
    private ActivitySplashScreenBinding activitySplashScreenBinding;
    private Boolean flag = false;
    private SessionManager sessionManager;
//    endregion

    @Override
    protected void onCreate(Bundle savedInstanceState)
    {
//       Inflate layout
        super.onCreate(savedInstanceState);
        activitySplashScreenBinding = DataBindingUtil.setContentView(this, R.layout.activity_splash_screen);

//        Set Fullscreen
        getWindow().getDecorView().setSystemUiVisibility(View.SYSTEM_UI_FLAG_FULLSCREEN
                |View.SYSTEM_UI_FLAG_HIDE_NAVIGATION);
//        Hooks
        sessionManager = new SessionManager(this);
//        Try again click listener
        activitySplashScreenBinding.tryAgain.setOnClickListener(view -> loading());
//        Check User logged in
        flag = sessionManager.isLoggedIn();
//        Check net connection
        checkInternet();
    }

//    region methods
//    Check Internet Connection
    private void checkInternet()
    {
        if (sessionManager.checkConnection())
        {
//            Set timer to go to next activity
            int SPLASH_SCREEN = 2400;

            if (flag)
            {
                new Handler().postDelayed(this::goToHomeActivity, SPLASH_SCREEN);
            }
            else
            {
                new Handler().postDelayed(this::goToRegLogActivity, SPLASH_SCREEN);
            }
        }
        else
        {
            activitySplashScreenBinding.tryAgainLinearLayout.setVisibility(View.VISIBLE);
            activitySplashScreenBinding.tryAgain.setVisibility(View.VISIBLE);
        }
    }
//    GoTo another activities
    private void goToRegLogActivity()
    {
        Intent regLogActivity = new Intent(SplashScreenActivity.this, AuthActivity.class);
        startActivity(regLogActivity);
        finish();
    }
    private void goToHomeActivity()
    {
        Intent homeActivity = new Intent(SplashScreenActivity.this, HomeActivity.class);
        startActivity(homeActivity);
        finish();
    }
//    Try again connection
    private void loading()
    {
        int TRY_AGAIN = 1000;
        activitySplashScreenBinding.tryAgain.startAnimation();
        new Handler().postDelayed(this::tryAgain, TRY_AGAIN);
    }
    @RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
    private void tryAgain()
    {
        if (sessionManager.checkConnection())
        {
//            Check User logged in
            if (flag)
            {
                goToHomeActivity();
            }
            else
            {
                goToRegLogActivity();
            }
        }
        else
        {
            activitySplashScreenBinding.tryAgain.revertAnimation();
        }
    }
//    endregion
}

File: app/src/main/java/ir/arapp/arappmain/view/activities/SplashScreenActivity.kt
package ir.arapp.arappmain.view.activities

import android.content.Intent
import android.os.Bundle
import android.os.Handler
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import ir.arapp.arappmain.R
import ir.arapp.arappmain.databinding.ActivitySplashScreenBinding
import ir.arapp.arappmain.model.base.*
import ir.arapp.arappmain.util.server.RetrofitClient
import ir.arapp.arappmain.util.services.SessionManager


class SplashScreenActivity : AppCompatActivity() {
    //    region Variables
    private var _activitySplashScreenBinding: ActivitySplashScreenBinding? = null
    private val activitySplashScreenBinding get() = _activitySplashScreenBinding!!
    private var flag = false
    private var sessionManager: SessionManager? = null

    //    endregion
    override fun onCreate(savedInstanceState: Bundle?) {
//       Inflate layout
        super.onCreate(savedInstanceState)
        _activitySplashScreenBinding =
            DataBindingUtil.setContentView(this, R.layout.activity_splash_screen)

//        Set Fullscreen
        window.decorView.systemUiVisibility = (View.SYSTEM_UI_FLAG_FULLSCREEN
                or View.SYSTEM_UI_FLAG_HIDE_NAVIGATION)
        //        Hooks
        sessionManager = SessionManager(this)
        //        Try again click listener
        activitySplashScreenBinding.tryAgain.setOnClickListener { view -> loading() }
        //        Check User logged in
        flag = sessionManager!!.isLoggedIn
        //        Check net connection
        checkInternet()
    }

    //    region methods
    //    Check Internet Connection
    private fun checkInternet() {
        if (sessionManager!!.checkConnection()) {
//            Set timer to go to next activity
            val SPLASH_SCREEN = 2400
            if (flag) {
                Handler().postDelayed({ goToHomeActivity() }, SPLASH_SCREEN.toLong())
            } else {
                Handler().postDelayed({ goToRegLogActivity() }, SPLASH_SCREEN.toLong())
            }
        } else {
            activitySplashScreenBinding!!.tryAgainLinearLayout.visibility = View.VISIBLE
            activitySplashScreenBinding!!.tryAgain.visibility = View.VISIBLE
        }
    }

    //    GoTo another activities
    private fun goToRegLogActivity() {
        val regLogActivity = Intent(this@SplashScreenActivity, AuthActivity::class.java)
        startActivity(regLogActivity)
        finish()
    }

    private fun goToHomeActivity() {
        RetrofitClient.login(sessionManager!!.getPhoneNumber(this)!!,sessionManager!!.getPassword(this)!!)
        val homeActivity = Intent(this@SplashScreenActivity, HomeActivity::class.java)
        startActivity(homeActivity)
        finish()
    }

    //    Try again connection
    private fun loading() {
        val TRY_AGAIN = 1000
        activitySplashScreenBinding!!.tryAgain.startAnimation()
        Handler().postDelayed({ tryAgain() }, TRY_AGAIN.toLong())
    }

    private fun tryAgain() {
        if (sessionManager!!.checkConnection()) {
//            Check User logged in
            if (flag) {

                goToHomeActivity()
            } else {
                goToRegLogActivity()
            }
        } else {
            activitySplashScreenBinding!!.tryAgain.revertAnimation()
        }
    } //    endregion
}